### PR TITLE
Add template provisioning

### PR DESCRIPTION
**Pull request by vfebvre**

### CODE REVIEW

File: app/models/manageiq/providers/ibm_power_hmc/infra_manager/provision.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ManageIQ::Providers::IbmPowerHmc::InfraManager::Provision < MiqProvision
+  include_concern 'Cloning'
+  include_concern 'StateMachine'
+  def destination_type
+    "Vm"
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_power_hmc/infra_manager/provision.rb:5]: If you're provisioning _to_ a template then the destination_type should be `"Template"

Comment (agrare) [reply app/models/manageiq/providers/ibm_power_hmc/infra_manager/provision.rb:5]: You should also validate that the request_type is `"clone_to_template"`, see https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/app/models/manageiq/providers/vmware/infra_manager/provision.rb#L9 for examples

**Comment by chessbyte** [comment]
@jaywcarman @agrare should this be part of an epic on the ManageIQ roadmap?  also, is there anything to update in the [Capabilities Matrix](https://www.manageiq.org/docs/reference/latest/capabilities_matrix)?

**Comment by agrare** [comment]
@chessbyte is it part of the https://github.com/ManageIQ/manageiq-providers-ibm_power_hmc/issues/8 enhancement tracking issue which is on the roadmap.  An update to the capabilities matrix would be good yes

**Pull request merged**